Update multiple Folders and Tasks

giard-alexandre opened the pull request:
Title says it all! I don't know how exactly you'd like to handle Integration tests for this as the only thing we can update when updating multiple Folders or Tasks are Custom Fields.

Open to suggestions if you don't want to merge without tests!

PR fixes: #26

agiard committed:
Update Readme with simple http client usage

giard-alexandre committed:
Merge pull request #1 from staviloglu/master

Update README.md

agiard committed:
Add UpdateTasksAsync
Allows tu update multiple tasks at once.

staviloglu committed:
Changed the <summary>

Changed the xml documentation comment <summary> to meet the [Wrike API documentation.](https://developers.wrike.com/api/v4/folders-projects/#modify-folder)

staviloglu commented on Taviloglu.Wrike.ApiClient/FoldersAndProjects/WrikeClient.FoldersAndProjects.cs:
```diff
@@ -157,6 +157,18 @@ async Task<WrikeFolder> IWrikeFoldersAndProjectsClient.UpdateAsync(WrikeClientId
         }
 
+        async Task<IEnumerable<WrikeFolder>> IWrikeFoldersAndProjectsClient.UpdateFoldersAsync(WrikeClientIdListParameter folderIds, List<WrikeCustomFieldData> customFields, List<string> optionalFields)
+        {
```
We should update [WrikeFolder.OptionalFields](https://github.com/staviloglu/Taviloglu.Wrike.ApiClient/blob/e42dda18a50a6c713df50d1ed3a9afc0907669fc/Taviloglu.Wrike.Core/FoldersAndProjects/WrikeFolder.cs#L152) to include all the values below
- metadata	Folder metadata entries
- hasAttachments	Has attachments
- attachmentCount	Attachment count
- description	Description
- briefDescription	Brief description
- customFields	Custom fields
- customColumnIds	Associated custom field IDs
- superParentIds	List of super parent folder IDs (applicable to 'Selective Sharing' labs feature)
- space	Is folder a space
- contractType	Contract type

and **allow only contractType** to be passed as parameter. Please see [GetFoldersAsync](https://github.com/staviloglu/Taviloglu.Wrike.ApiClient/blob/e42dda18a50a6c713df50d1ed3a9afc0907669fc/Taviloglu.Wrike.ApiClient/FoldersAndProjects/WrikeClient.FoldersAndProjects.cs#L25) which can be taken as sample for optionalFields validation

staviloglu committed:
Changed the <summary>

Changed the xml documentation comment <summary> to meet the [Wrike API documentation.](https://developers.wrike.com/api/v4/tasks/#modify-tasks)

staviloglu committed:
Changed the uri and removed optional fields"

Client should send the request to "/API/V4/TASKS/{taskId},{taskId}" 
There is no optionalFields parameter in the [documentation] (https://developers.wrike.com/api/v4/tasks/#modify-tasks)

staviloglu commented:
Hi @heuristicAL 

Thanks for the PR to fix the missing parts in the client. 

I have directly committed into your branch to fix some code, please see them. 

And also I have the comment above about the WrikeFolder.OptionalFields

I think we can write integration test method to see if the customField of the folder / task is updated or not.
It could be like;
1. Create new task/folder add customeField
2. Update the customField
3. Check if it is updated
4. Delete the newly create task/folder.

I can add some unit tests later

staviloglu commented:
@heuristicAL thx for the PR
I'm implementing these multi-update methods in another [branch](https://github.com/staviloglu/Taviloglu.Wrike.ApiClient/commit/ca9f6d54735836835b039a7683e1be52b4a3d6ec) so, closing the PR.
client v3.0 will have these features.
